### PR TITLE
[NEW-UI]: Create and update task

### PR DESCRIPTION
**Pull request by zankevich**

### CODE REVIEW

**Comment by nmanovic** [comment]
@zankevich, Is the PR in progress?

**Comment by zankevich** [comment]
> @zankevich, Is the PR in progress?

Yes

**Comment by bsekachev** [comment]
@zankevich 
This PR is ready?

**Comment by zankevich** [comment]
> @zankevich
> This PR is ready?

Yes

**Comment by bsekachev** [comment]
Such a size of the dialog is expected?
Screenshot from 2019-08-13 15-50-55

**Comment by zankevich** [comment]
> @zankevich
> Such a size of the dialog is expected?
> Screenshot from 2019-08-13 15-50-55

Yes, can change to any size if needed.

**Comment by bsekachev** [comment]
@zankevich 
When I click the button "Create", the dialog closes and nothing happens.
It is not obvious. User doesn't know whether files were loaded and can leave from the tab. In such case task creating will be canceled. User doesn't know creation status, or whether his task was created or some errors occurred.

**Comment by bsekachev** [comment]
@zankevich 
Share works incorrect. 
The directory structure is inconsistent.
I just attach an example.
Screenshot from 2019-08-13 16-03-35

**Comment by bsekachev** [comment]
@zankevich 
Let's remove the input 'Dataset repository'. 
It is a plugin functionality and shouldn't be hardcoded. 
We haven't mentioned about it before. Sorry about that.

File: cvat-ui/src/components/modals/task-create/task-create.tsx [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+  private getSharedFiles = (directory: string) => {
+    return this.store.dispatch(getShareFilesAsync(directory));
+  }
+
+  private onTreeNodeSelect = (keys: any, event: any) => {
+    console.log('Trigger Select', keys, event);
+  }
+
+  private onTreeNodeExpand = () => {
+    console.log('Trigger Expand');
```
Comment (bsekachev) [inline cvat-ui/src/components/modals/task-create/task-create.tsx]: Why do we need these stubs?

File: cvat-ui/src/utils/tasks-dto.ts [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+  }
+
+  callback();
+}
+
+export function convertStringToNumber(event: React.FormEvent<HTMLInputElement>) {
+  const convertedValue = Number(event.currentTarget.value);
+
+  if (isNaN(convertedValue)) {
+    return 0;
```
Comment (bsekachev) [inline cvat-ui/src/utils/tasks-dto.ts]: It can return 0 for segment_size and image_quality if I enter a string. And validator says nothing, right?

Comment (zankevich) [reply cvat-ui/src/utils/tasks-dto.ts]: @bsekachev Is this incorrect?

**Comment by zankevich** [comment]
> @zankevich
> Let's remove the input 'Dataset repository'.
> It is a plugin functionality and shouldn't be hardcoded.
> We haven't mentioned about it before. Sorry about that.

Removed

**Comment by zankevich** [comment]
> @zankevich
> Share works incorrect.
> The directory structure is inconsistent.
> I just attach an example.
> Screenshot from 2019-08-13 16-03-35

Fixed

**Comment by zankevich** [comment]
> @zankevich
> When I click the button "Create", the dialog closes and nothing happens.
> It is not obvious. User doesn't know whether files were loaded and can leave from the tab. In such case task creating will be canceled. User doesn't know creation status, or whether his task was created or some errors occurred.

Fixed

File: cvat-ui/src/utils/tasks-dto.ts (hunk 2 above)
Comment (bsekachev) [reply cvat-ui/src/utils/tasks-dto.ts]: @zankevich 
0 is out of range for these inputs

**Comment by bsekachev** [comment]
@zankevich
Still have a problem with task creating. If I enter something wrong to the fields "segment size" and "overlap size", the dialog doesn't allow create a task, even if I try to remove entered data. 
Screenshot from 2019-08-15 17-31-34

File: cvat-ui/src/utils/tasks-dto.ts (hunk 2 above)
Comment (bsekachev) [reply cvat-ui/src/utils/tasks-dto.ts]: UPD: I noticed, image quality can be 0

**Pull request merged**